fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java

```diff
@@ -495,32 +490,112 @@ static void checkOperateLoadPrivilege(ConnectContext context, String dbName, Str
      * check privilege by AST tree
      */
     private static class PrivilegeCheckerVisitor extends AstVisitor<Void, ConnectContext> {
+
         public void check(StatementBase statement, ConnectContext context) {
             visit(statement, context);
         }
 
+        // --------------------------------- Query Statement -------------------------------------
+
         @Override
-        public Void visitDeleteStatement(DeleteStmt statement, ConnectContext context) {
-            checkTableAction(context, statement.getTableName(), PrivilegeType.DELETE);
+        public Void visitQueryStatement(QueryStatement statement, ConnectContext context) {
+            Map<TableName, Table> allTouchedTables = AnalyzerUtils.collectAllTableAndView(statement);
+            for (Map.Entry<TableName, Table> tableToBeChecked : allTouchedTables.entrySet()) {
+                TableName tableName = tableToBeChecked.getKey();
+                Table table = tableToBeChecked.getValue();
+
+                if (table instanceof View) {
+                    if (!PrivilegeActions.checkViewAction(context, tableName.getDb(), tableName.getTbl(), PrivilegeType.SELECT)) {
+                        ErrorReport.reportSemanticException(ErrorCode.ERR_PRIVILEGE_ACCESS_TABLE_DENIED,
+                                context.getQualifiedUser(), tableName);
+                    }
+                } else if (table instanceof SchemaTable && ((SchemaTable) table).isBeSchemaTable()) {
+                    checkStmtOperatePrivilege(context);
```

Comment:
the original logic is that  we want to check both select and operate priv on be schema table

Reply:
OK